### ISSUE TITLE
[java] LocalVariableCouldBeFinal false positive in exception handling

### ISSUE THREAD

**Issue by djh82** [pmd/pmd#5049]
**Affects PMD Version:** 7.2

**Rule:** LocalVariableCouldBeFinal

Please provide the rule name and a link to the rule documentation:
<https://docs.pmd-code.org/latest/pmd_rules_java_codestyle.html#localvariablecouldbefinal>

**Description:**
False positive; compiler error if applying suggestion that 'local variable might already have been assigned'.

**Code Sample demonstrating the issue:**

```java
public void test() {
    Object o;
    try {
        o = new Object();
    } catch (Exception e) {
        o = new Object();
    }
}
```

**Expected outcome:**

PMD shouldn't warn on this.

**Running PMD through:** CLI

**Comment by oowekyala** [comment]
Duplicate of #5046
